### PR TITLE
moves cloud providers to its own pkg

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply moves cloud providers code to `handler/v1/provider` pkg.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/v1/common/error.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package common
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/common/error.go:1]: extracted from `request.go` file

File: api/pkg/handler/v1/common/request.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
+package common
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/common/request.go:1]: moved from `request.go` file

File: api/pkg/handler/v1/provider/digitalocean.go [hunk 3]
```diff
@@ -87,3 +89,33 @@ func digitaloceanSize(ctx context.Context, token string) (apiv1.DigitaloceanSize
 
 	return sizeList, nil
 }
+
+// DoSizesNoCredentialsReq represent a request for digitalocean sizes EP,
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/provider/digitalocean.go:93]: types moved from `request.go` file

File: api/pkg/handler/v1/provider/openstack.go [hunk 4]
```diff
@@ -328,14 +330,118 @@ func getClusterForOpenstack(ctx context.Context, projectProvider provider.Projec
...
-		return nil, kubernetesErrorToHTTPError(err)
+		return nil, common.KubernetesErrorToHTTPError(err)
 	}
 	if cluster.Spec.Cloud.Openstack == nil {
 		return nil, errors.NewNotFound("cloud spec for ", clusterID)
 	}
 	return cluster, nil
 }
+
+// OpenstackReq represent a request for openstack
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/provider/openstack.go:345]: types moved from request.go file

File: api/pkg/handler/v1/provider/vsphere.go [hunk 5]
```diff
@@ -79,3 +81,36 @@ func getVsphereNetworks(providers provider.CloudRegistry, username, password, da
 
 	return apiNetworks, nil
 }
+
+// VSphereNetworksReq represent a request for vsphere networks
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/provider/vsphere.go:85]: types moved from request.go file

File: api/pkg/handler/v1/provider/vsphere_test.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package handler_test
+package provider_test
```
Comment (p0lyn0mial) [inline api/pkg/handler/v1/provider/vsphere_test.go:1]: yay :)

**Comment by p0lyn0mial** [comment]
/hold 
because I want to test it manually.

**Review by zreigz — APPROVED** [review]
/lgtm
/approve

**Pull request merged**